bytecomp/translmod.ml

```diff
@@ -673,8 +657,32 @@ and transl_structure loc fields cc rootpath final_env = function
                transl_module Tcoerce_none None modl, body),
           size
 
+      | Tstr_open od ->
+          (* this optimization shouldn't be needed because Simplif would
+             actually remove the [Llet] when it's not used.
+             But since [scan_used_globals] runs before Simplif, we need to do
+             it. *)
+          begin match od.open_bound_items with
```

Comment:
This isn't quite right. You also need to check that `pure_module od.open_expr` is `Alias`. Otherwise you may drop side effects for things like:

```ocaml
open struct let () = print_endline "Hello" end
```